metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage.py

```diff
@@ -299,58 +363,71 @@ def _get_upstream_lineage_info(
                 self.report.upstream_lineage[dataset_name] = [
                     u.dataset for u in upstream_tables
                 ]
-            return (
-                UpstreamLineage(
-                    upstreams=upstream_tables,
-                    fineGrainedLineages=sorted(
-                        finegrained_lineages, key=lambda x: (x.downstreams, x.upstreams)
-                    )
-                    or None,
-                ),
-                column_lineage,
+            return UpstreamLineage(
+                upstreams=upstream_tables,
+                fineGrainedLineages=sorted(
+                    finegrained_lineages, key=lambda x: (x.downstreams, x.upstreams)
+                )
+                or None,
             )
+
         return None
 
     def _populate_view_lineage(self, conn: SnowflakeConnection) -> None:
         with PerfTimer() as timer:
             self._populate_view_upstream_lineage(conn)
             self.report.view_upstream_lineage_query_secs = timer.elapsed_seconds()
-        with PerfTimer() as timer:
-            self._populate_view_downstream_lineage(conn)
-            self.report.view_downstream_lineage_query_secs = timer.elapsed_seconds()
+        if self.report.edition == SnowflakeEdition.STANDARD:
```

Comment:
just a question - don't we already cover this case with the check above? line 187

Reply:
I think for the populate_lineage case we are doing this check in the caller method. For consistency, why don't we put this in one place? Is it possible to consolidate, or make this consistent?

Reply:
Snowflake Standard Edition 

- supports "Table to View" Lineage via `snowflake.account_usage.object_dependencies` view 
- supports "S3 to Table" Lineage via `show external tables` queries.
- does not support "View to Table" Lineage via `snowflake.account_usage.access_history` view
- does not support "Table to Table" Lineage via `snowflake.account_usage.access_history` view
- does not support "S3 to Table" Lineage via `snowflake.account_usage.access_history` view

Hence the check is present at different places, based on which type of lineage type is being extracted.

Reply:
Okay - this is _extremely useful_ context. Can we add this as a comment into the source itself to keep this knowledge?

Reply:
done